common/app-starters-postprocessor-common/src/main/java/org/springframework/cloud/stream/app/postprocessor/SecurityDisabledEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,96 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.postprocessor;
+
+import java.util.Properties;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.util.StringUtils;
+
+/**
+ * With SpringBoot 2+ the security is enabled by default. To disable it one can exclude the
+ * {@link org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration} using the
+ * `spring.autoconfigure.exclude` property. For example:
+ * (--spring.autoconfigure.exclude=org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration)
+ *
+ * Conversely if the SecurityAutoConfiguration is already excluded, the security can be re-enabled by overriding and
+ * removing the SecurityAutoConfiguration form the spring.autoconfigure.exclude.
+ *
+ * {@link SecurityDisabledEnvironmentPostProcessor} allows controlling the security enabled/disabled behavior with the help of a
+ * dedicated boolean property: `spring.cloud.security.disabled`.
+ *
+ *  - If the `spring.cloud.security.disabled` property is not used the {@link SecurityDisabledEnvironmentPostProcessor} is ignored.
+ *
+ *  - If the `spring.cloud.security.disabled` (or `spring.cloud.security.disabled=true`) is set then the result behavior is equivalent
+ * to: spring.autoconfigure.exclude=org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration
+ *
+ *  - If the `spring.cloud.security.disabled=false` is used the result behavior is equivalent to removing spring.autoconfigure.exclude
+ *  from the property source.
+ *
+ *
+ *  Note: the PostProcessor will not alter any spring.autoconfigure.exclude set outside the scope of the post-processor.
+ *
+ * @author Christian Tzolov
+ */
+public class SecurityDisabledEnvironmentPostProcessor implements EnvironmentPostProcessor {
+
+	public static final String SECURITY_DISABLED_PROPERTY = "spring.cloud.security.disabled";
+	protected static final String PROPERTY_SOURCE_KEY_NAME = SecurityDisabledEnvironmentPostProcessor.class.getName();
+	protected static final String AUTO_CONFIGURE_EXCLUDE_PROPERTY = "spring.autoconfigure.exclude";
+	protected static final String SECURITY_AUTO_CONFIGURATION_CLASS = "org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration";
+
+	public SecurityDisabledEnvironmentPostProcessor() {
+		super();
+	}
+
+	@Override
+	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication springApplication) {
+
+		// If the SECURITY_DISABLED_PROPERTY is not used, ignore the post processor al together.
```

Comment:
That's something wrong: I though that idea is to make apps unsecured by default.

I think you will believe more to @sabbyanandan  😄

Reply:
I know that the policies are not streamlined yet and this one of the reasons to mark it as `do not merge` :)

I've run through a several permutations of those policies and still not sure what is best. 
For example it took me some time to realize that we can use the post-processor's property-source itself as an internal (hidden) state to distinct between `exclusions` made by the post-processor vs. external such. Later allows to give precedence to the external ones.  Still not sure this logic is effect free though. 
Also we need to decide how this post-processor is going to be applied to all app starters.

Reply:
This is outdated already and resolved fully different way.
So, closing conversation...